jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -249,9 +265,35 @@ private SpanContext createNewContext(String debugId) {
       return new SpanContext(id, id, 0, flags);
     }
 
-    private SpanContext createChildContext(SpanContext preferredParent) {
+    private Map<String, String> createChildBaggage() {
+      Map<String, String> baggage = null;
+
+      // optimization for 99% use cases, when there is only one parent
+      if (references.size() == 1) {
+        return references.get(0).getSpanContext().baggage();
+      }
+
+      for (Reference reference: references) {
+        if (reference.getSpanContext().baggage() != null) {
+          if (baggage == null) {
+            baggage = new HashMap<String, String>();
+          }
+          baggage.putAll(reference.getSpanContext().baggage());
+        }
+      }
+
+      return baggage;
+    }
+
+    private SpanContext createChildContext() {
+      if (references.isEmpty()) {
```

Comment:
nit: I think we can do away with this check, as well as the similar check in preferredParent(), because both of these methods are only called by start() when `!references.isEmpty()`. And if by accident it gets called otherwise, it will throw exception from `references.get(0);` in preferredParent() anyway.